rewrite-javascript/rewrite/src/javascript/tree.ts

```diff
@@ -643,6 +644,21 @@ export namespace JS {
         readonly type?: JavaType;
     }
 
+    /**
+     * Represents function calls which are not method invocations.
+     * @example f(5, 0, 4)
+     * @example f?.(5, 0, 4)
+     * @example data["key"](5, 0, 4)
+     * @example (() => { return 3 + 5 })()
+     */
+    export interface FunctionCall extends JS, Expression {
+        readonly kind: typeof Kind.FunctionCall;
+        readonly select?: J.RightPadded<Expression>;
```

Comment:
I think we should instead use something like `function`.
```suggestion
        readonly function?: J.RightPadded<Expression>;
```

Also, to clarify, is the `J.RightPadded` required for use cases like `f?.(1)` where the padding would be what is between `f` and `?.`?

Reply:
> I think we should instead use something like function.

3b998be37

Reply:
Re padding:
Currently (9c8d727789ce4d74e379e2de9777e26a75a6b143) the code:
```
func/*a*/?./*b*/("TS")
```
gets parsed as:
```
FunctionCall(function=func, markers = Optional(prefix=/*a*/), arguments=Container(prefix=/*b*/)
```

For comparison, the code with method invocation:
```
console/*e*/./*f*/log("a")
```
is parsed as:
```
MethodInvocation(select=(Identifier(e), after= /*e*/), name=(Identifier(log), before=/*f*/)
```

Reply:
After 5fb8c1d

It's:

| code | parsed as |
| ------| ----| 
| `func/*a*/?./*b*/("TS")` | `FunctionCall(function=(func, after=/*a*/, markers = Optional(prefix=empty), arguments=Container(prefix=/*b*/))` |
| `func/*a*/?./*b*/call("TS")` | `MethodInvocation(select=(func, after=/*a*/), markers = Optional(prefix=empty), name=(Identifier(call), prefix=/*b*/))` |
| `func/*a*/.call/*b*/("TS")` |  `MethodInvocation(select=(func, after=/*a*/), markers = empty, name = (Identifier(call), prefix=/*b*/))` |